src/build/arg/settings.rs

```diff
@@ -39,7 +39,7 @@ impl_settings! { ArgSettings, ArgFlags,
     Required("required") => Flags::REQUIRED,
     MultipleOccurrences("multipleoccurrences") => Flags::MULTIPLE_OCC,
     MultipleValues("multiplevalues") => Flags::MULTIPLE_VALS,
-    AllowEmptyValues("allowemptyvalues") => Flags::EMPTY_VALS,
+    NoEmptyValues("noemptyvalues") => Flags::NO_EMPTY_VALS,
```

Comment:
I don't think we need to rename this. We can work with setting always on.

Reply:
With `AllowEmptyValues` on by default. If the users wants a arg doesn't accepts empty values, he needs to unset `AllowEmptyValues` then set `TakesValue`, which is counterintuitive.....

Reply:
`TakesValue` shouldn't be unset when `AllowEmptyValues` is unset.

Reply:
So #2233 related.

Reply:
Well, another problem is `AllowEmptyValues` requires `takes_value`. If `AllowEmptyValues` by default, all args that doesn't set `takes_value` won't pass debug_assertion. And I don't think removing this requirement is good.....

Reply:
Okay, I want to see how it looks with what you are proposing. And change the previous (now deleted) `allow_empty_values` function to `forbid_empty_values`.

Reply:
Done.